### ISSUE TITLE
Question about XdawnCovariances

### ISSUE THREAD

**Issue by ibr2011** [pyRiemann/pyRiemann#201]
Greetings, 

I hope all is well with you. 

I have a question about the parameter nfilter and what does it represent. Does it represent the number of classes? Or it could be any number?

The thing is every time I increase the number of nfilter the performance of my model increases and I'm not sure this is the correct way.

Kind regards,
Ibrahim

**Comment by sylvchev** [comment]
Hi @ibr2011 
I assume that you talk about the argument `nfilter` used by XDAWN algorithm, you could read the paper to have detail : https://doi.org/10.1109/TBME.2009.2012869 
XDAWN objectives are to enhance and refine signal of interest. It is often used to reduce the input dimension (number of electrodes for EEG) for keeping the computation tractable. Broadly speaking, you could think as `nfilter` as a somewhat to the number of components of ICA or PCA. It is possible that you obtain higher classfication performances with higher `nfilter` values, the maximum `nfilter` value being the dimension of input signal (number of electrodes for EEG). IMO, it depends mostly to the EEG recording quality.

**Comment by qbarthelemy** [comment]
As said previously, `nfilter` will define the dimension of filtered/reduced/enhanced data.
A high number of components leads to an overfitting risk: it increases performances on training set, but decreases them on test set. `nfilter` is usually chosen between 2 and 4. If it is necessary, you can apply a cross-validation to set `nfilter`.

**Comment by ibr2011** [comment]
Thank you @qbarthelemy and @sylvchev for your valuable comments. 

Keep up with the good work. 

Kind regards,
Ibrahim
